sessionctx/variable/statusvar.go

```diff
@@ -49,6 +49,13 @@ func RegisterStatistics(s Statistics) {
 	statisticsListLock.Unlock()
 }
 
+// ClearStatistics4Test clear statistics list for test
+func ClearStatistics4Test() {
```

Comment:
It seems before this PR, we don't need this function, could you elaborate a bit why you add this function?

Reply:
I like it. It is testing the statistics that are exposed through `SHOW GLOBAL STATUS`. The `Clear` ensures that it is not flaky from other tests.

MySQL actually has [`FLUSH STATUS`](https://dev.mysql.com/doc/refman/8.0/en/flush.html#flush-status) to reset the statistics. We don't implement it (the syntax is a noop). If you can rename this to `FlushStatistics()`, it will be easy enough to add correct support for `FLUSH STATUS` in another PR. Here's the relevant code:

https://github.com/pingcap/tidb/blob/cdd8fc48aa02be86980e19a82d9f61b90b8d11f7/executor/simple.go#L1492-L1527

Reply:
Thanks for your information @morgo . That sounds good 😄

Reply:
Thanks for your review @tisonkun @morgo.  
The function to get StatusVar is shared globally ("show status" statements).
https://github.com/pingcap/tidb/blob/1c2437e2ea4df2b3491d53a8a0d9a83757e5d8d3/sessionctx/variable/statusvar.go#L25-L26
https://github.com/pingcap/tidb/blob/1c2437e2ea4df2b3491d53a8a0d9a83757e5d8d3/sessionctx/variable/statusvar.go#L45-L50
https://github.com/pingcap/tidb/blob/1c2437e2ea4df2b3491d53a8a0d9a83757e5d8d3/sessionctx/variable/statusvar.go#L52-L59
 In every unit test, as long as we use the testkit.CreateMockStoreAndDomain method (new store, leading to new domain, leading to new ddl), we will register a statistics interface into the list in the statusvar package. 

https://github.com/pingcap/tidb/blob/1c2437e2ea4df2b3491d53a8a0d9a83757e5d8d3/ddl/ddl.go#L384-L390

However, there is no way to revoke the instance. When the unit test is over, the domain object and its associated ddl object are closed, and its object still erroneously exists in the list of the statusvar package. When the next unit test, if the "show status" statement is executed, it will traverse to the closed objects, resulting in an error that the rpc client has been closed(test store is closed already).
https://github.com/pingcap/tidb/blob/1c2437e2ea4df2b3491d53a8a0d9a83757e5d8d3/sessionctx/variable/statusvar.go#L59-L63
The simple process is shown in the chart below.

![image](https://user-images.githubusercontent.com/12460574/133026012-7ef4cb3c-89f4-46f4-a69f-64c4be74ec9d.png)
[image](https://user-images.githubusercontent.com/12460574/133029421-8ccd6679-ebb1-4020-b02e-51af6b7813bf.png)

Reply:
ClearStatistics4Test is only used to clear the list of statistics interface in the test, to ensure that it will not receive side effect from other tests. It cannot be used for "flush status". After executing this function, all instances of actual execution statistics are cleared. After this, any status  can no longer be obtained (the list is empty).
The code block is shown below.
https://github.com/pingcap/tidb/blob/1c2437e2ea4df2b3491d53a8a0d9a83757e5d8d3/sessionctx/variable/statusvar.go#L59-L63

Reply:
OK. Thank you for the reply.

Reply:
Thanks to their great work https://github.com/pingcap/tidb/pull/28027, there is no need to add this function `ClearStatistics4Test()`.
So I remove my unnessary code.

Reply:
That's a good news. Interesting collarboration.